### ISSUE TITLE
Dialog: Default value of "hidden" prop is true, should be false according to Docs/Types

### ISSUE THREAD

**Issue by Nimelrian** [OfficeDev/office-ui-fabric-react#3387]
### Bug Report

- __Package version(s)__:
  - office-ui-fabric-react: 5.20.3
  - react: 16.0.0
  - react-dom: 16.0.0
- __Browser and OS versions__: Chrome Version 62.0.3202.89 / Windows 10 Pro Version 1703

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Normal

#### Describe the issue:

Rendering a Dialog without setting the hidden prop causes it to [default to true](https://github.com/OfficeDev/office-ui-fabric-react/blob/227159e1551465b032db4bdddbcf65bc7d54248f/packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx#L30). According to the docs and [typings](https://github.com/OfficeDev/office-ui-fabric-react/blob/227159e1551465b032db4bdddbcf65bc7d54248f/packages/office-ui-fabric-react/src/components/Dialog/Dialog.types.ts#L31-L35) it should be false instead.

#### Actual behavior:

Rendering a Dialog without the hidden prop set to any value causes it to be hidden.

#### Expected behavior:

Rendering a Dialog without the hidden prop set to any value causes it to be rendered normally.

### If applicable, please provide a codepen repro:

CodeSandbox instead of codepen, but does the same job:

https://codesandbox.io/s/9o3r83zlxo

**Comment by codefactorymx** [comment]
On components/dialog/dialog.tsx find this code
isOpen={ isOpen !== undefined ? isOpen : !hidden }

Also on the properties is the value of the hidden

**Comment by derek-smith** [comment]
I just wasted an hour trying to figure out why my Dialog wasn't rendering. Please update the documentation or the code to match.

**Comment by martellaj** [comment]
@dzearing @joschect Is there someone who owns this component right now? Pretty bad bug...

**Comment by joschect** [comment]
It looks like this has been this way for a long while. ITT this isn't a bug and a case of incorrect documentation @micahgodbolt might know more.

**Comment by micahgodbolt** [comment]
Yeah, we definitely need to get this sorted out. I "believe" that the code is correct and the documentation is wrong. The deprecated 'isOpen' has a default of false, as well as the Modal 'isOpen' defaults to false. So the intention is that modals do not display by default (and are typically controlled components anyway). 

If we make a change, I'd like to see both Modal and Dialog updated to being the same. Dialog is just a specific variant of a modal.

**Comment by micahgodbolt** [comment]
but in the same breath, props are typically false by default and enabled by setting them to true. So i'm even wondering why we moved away from 'isOpen' which is a prop you can set to true to change the default state. 

@dzearing. Was there a reason we opted to use 'hidden' instead of 'isOpen'? Or did I deprecate the wrong prop (very possible).

**Comment by Nimelrian** [comment]
Any news on this? This is still present as of 5.80.0 https://codesandbox.io/s/vw1l5l5w7

Docs are also still saying that hidden is false by default.
